src/rgw/rgw_rest.cc

```diff
@@ -779,6 +779,29 @@ void dump_range(struct req_state *s, uint64_t ofs, uint64_t end,
 
 int RGWGetObj_ObjStore::get_params()
 {
+  // start gettorrent
+  const char *query_str = s->info.env->get("QUERY_STRING");
+  if (0 == strcmp(query_str, GET_TORRENT))
+  {
+    if ((g_conf->torrent_tracker).empty() && (g_conf->torrent_origin).empty())
+    {    
+      INFO "add torrent_tracker and torrent_origin into ceph.conf" << dendl;
+      return -1;
+    }    
+
+    torrent.set_flag(true);
+    torrent.set_info_piece_length(g_conf->sha_unit);
+    string comment(query_str);
+    torrent.set_comment(g_conf->torrent_comment);
+    torrent.set_create_by(g_conf->torrent_createby);
+    torrent.set_announce(g_conf->torrent_tracker);
+    torrent.set_encoding(g_conf->torrent_encoding);
+    torrent.set_origin(g_conf->torrent_origin);
+    string name = obj.get_orig_obj();
+    torrent.set_info_name(name);
```

Comment:
i think that using `get_orig_obj()` to generate the torrent object names will prevent us from supporting object versioning and object instances. if you overwrite the original object with new data, you'd still find and return the old torrent object for it

i'd lean towards using the omap for this storage so that it's tied to a specific object version/instance, and gets cleaned up automatically when the object is deleted